pds_doi_core/input/osti_input_validator.py

```diff
@@ -0,0 +1,48 @@
+#!/bin/python
+#
+#  Copyright 2020, by the California Institute of Technology.  ALL RIGHTS
+#  RESERVED. United States Government Sponsorship acknowledged. Any commercial
+#  use must be negotiated with the Office of Technology Transfer at the
+#  California Institute of Technology.
+#
+# ------------------------------
+
+from lxml import etree
+from lxml import isoschematron
+
+from pds_doi_core.util.general_util import get_logger
+
+logger = get_logger('pds_doi_core.input.osti_input_util')
+
+class OSTIInputValidator:
```

Comment:
I am picky, but in the logic of a validator object the schematron should be instanciated as an attribute of the object in the __init__()
That would be this line:
schematron = isoschematron.Schematron(sct_doc, store_report=True)

This way the object can revalidate many document from the same schematron

Reply:
I am seeing that the date can not be validated with schematron and python. So that is good that you did it in the code.

To keep things simple I would not check for the status or extraneous fields if they don't break OSTI service. That will make the code more simple and will give us more flexibility in the future.

Reply:
Can indeed confirm that OSTI does not like any extra tags in the 'records' tag.

Extra tag added to test:

<records total="1">
    <record>
        <title>InSight Cameras Bundle 2020-07-27-10-42</title>
        <authors>
             <author>
                <first_name>R.</first_name>


Error message from OSTI:

pds_doi_core.actions.actions:run osti_response_str <records total="2" errors="1">
  <record status="Error" index="1">
    <site_code>NASA-PDS</site_code>
    <errors>
      <error>Title is required.</error>
      <error>A publication date is required.</error>
      <error>A site URL is required.</error>
      <error>A product type is required.</error>
      <error>A specific product type is required for non-dataset types.</error>
    </errors>
  </record>

Reply:
For some reason, the text above didn't get translated in the message, perhaps the extra greater than symbol:
It says:

records total="1"

which causes OSTI to barf.

Reply:
Ok, then may be we will use a schema xsd in addition. I don't like very much to have too much code for verification since we are supposed to work with an external file (schematron). We will see ho this can be improved. Jordan told me Ron is a schematron expert. Maybe he will be able to help us.